src/traces/scatter/plot.js

```diff
@@ -419,9 +419,6 @@ function plotOne(gd, idx, plotinfo, cdscatter, cdscatterAll, element, transition
         var enter = join.enter().append('path')
             .classed('point', true);
 
-        enter.call(Drawing.pointStyle, trace)
```

Comment:
@etpinard after thinking a little bit about it. I'm still think that translatePoints and poinStyle should be applied to the merge selection to keep in sync DOM and trace.